lib/dat-worker-pool/worker.rb

```diff
@@ -42,9 +42,12 @@ def prepend_after_work(&block);     self.after_work_callbacks.unshift(block);
 
     module InstanceMethods
 
-      def initialize(runner, queue)
+      attr_reader :dwp_number
+
+      def initialize(runner, queue, number)
         @dwp_runner  = runner
         @dwp_queue   = queue
+        @dwp_number  = number
```

Comment:
@jcredding maybe `@dwp_runner, @dwp_queue, @dwp_number = runner, queue, number` instead (less lines I guess)?  Don't really care either way - just kinda jumped out to me.